homeassistant/components/switch/broadlink.py

```diff
@@ -0,0 +1,261 @@
+    vol.Optional(CONF_COMMAND_ON, default=None): cv.string,
+    vol.Optional(CONF_FRIENDLY_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_OPTIMISTIC, default=True): cv.boolean,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SWITCHES): vol.Schema({cv.slug: SWITCH_SCHEMA}),
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_MAC): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Broadlink switches."""
+    import binascii
+    devices = config.get(CONF_SWITCHES, {})
+    switches = []
+    ip_addr = (config.get(CONF_HOST), 80)
+    mac_addr = binascii.unhexlify(
+        config.get(CONF_MAC).encode().replace(b':', b''))
+
+    for object_id, device_config in devices.items():
+
+        switches.append(
+            BroadlinkSwitch(
+                hass,
+                device_config.get(CONF_FRIENDLY_NAME, object_id),
+                device_config.get(CONF_COMMAND_ON),
+                device_config.get(CONF_COMMAND_OFF),
+                device_config.get(CONF_OPTIMISTIC),
+                ip_addr,
+                mac_addr
+            )
+        )
+
+    if not switches:
+        _LOGGER.error("No switches added.")
+        return False
+
+    add_devices(switches)
+
+
+class BroadlinkSwitch(SwitchDevice):
+    """Representation of an Broadlink switch."""
+
+    def __init__(self, hass, friendly_name, command_on,
+                 command_off, optimistic, ip_addr, mac_addr):
+        """Initialize the switch."""
+        self._hass = hass
+        self._name = friendly_name
+        self._state = False
+        self._command_on = command_on
+        self._command_off = command_off
+        self._optimistic = optimistic
+        self._ip_addr = ip_addr
+        self._mac_addr = mac_addr
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def assumed_state(self):
+        """Return if the state is based on assumptions."""
+        return self._optimistic
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        self._sendpacket(self._command_on)
+        self._state = True
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        self._sendpacket(self._command_off)
+        self._state = False
+
+    def _sendpacket(self, packet):
+        """Send packet to device."""
+        import base64
+        try:
+            broadlink = Broadlink.Device(self._ip_addr, self._mac_addr)
+            auth = broadlink.auth()
+            if auth:
+                _LOGGER.info('Packet sent successfully to device.')
+                broadlink.send_data(base64.b64decode(packet))
+
+        except ValueError as error:
+            _LOGGER.error(error)
+
+
+class Broadlink():
```

Comment:
Please extract this into a third party library. Home Assistant should not contain any protocol specific info.